init.lua

```diff
@@ -634,7 +650,8 @@ local function on_player_receive_fields(player, fields, update_callback)
 	if fields.teleport
 		and category == LOCATION_CATEGORY
 		and valid_entry_selected
-		and minetest.check_player_privs(player_name, "teleport") then
+		and (minetest.check_player_privs(player_name, "teleport")
+		or (personal_log_teleport_privilege and minetest.check_player_privs(player_name, "personal_log_teleport"))) then
```

Comment:
This would be best moved to a local helper function so that `make_personal_log_formspec` is guaranteed to check against the same conditions (no out-of-sync code issues).

Reply:
Done.